app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt

```diff
@@ -163,15 +171,73 @@ class NewMessageFragment : Fragment() {
     }
 
     private fun initDraftAndUi() {
-        newMessageViewModel.initDraftAndUi(newMessageActivityArgs)
-            .observe(viewLifecycleOwner) { isSuccess ->
-                if (isSuccess) {
-                    observeContacts()
-                    populateUiWithExistingDraftData()
-                } else {
-                    requireActivity().finish()
-                }
+        newMessageViewModel.initDraftAndUi(newMessageActivityArgs).observe(viewLifecycleOwner) { isSuccess ->
+            if (isSuccess) {
+                observeContacts()
+                populateUiWithExistingDraftData()
+                handleActionSend()
+            } else {
+                requireActivity().finish()
+            }
+        }
+    }
+
+    private fun handleActionSend() {
+        when (requireActivity().intent?.action) {
+            Intent.ACTION_SEND -> handleSingleSendIntent()
+            Intent.ACTION_SEND_MULTIPLE -> handleMultipleSendIntent()
+        }
+    }
+
+    /**
+     * Handle `Mailto` from [Intent.ACTION_VIEW] or [Intent.ACTION_SENDTO]
+     * Get [Intent.ACTION_VIEW] data with [MailTo] and [Intent.ACTION_SENDTO] with [Intent]
+     */
+    private fun handleMailTo() = with(binding) {
+        val intent = requireActivity().intent
+        intent?.data?.let { uri ->
+            if (!MailTo.isMailTo(uri)) return@with
+
+            val mailTo = MailTo.parse(uri)
+            val to = mailTo.to?.split(",")?.map { it.trim() } ?: emptyList()
+            val cc = mailTo.cc?.split(",") ?: intent.getStringArrayExtra(Intent.EXTRA_CC)?.toList() ?: emptyList()
+            val bcc = mailTo.bcc?.split(",") ?: intent.getStringArrayExtra(Intent.EXTRA_BCC)?.toList() ?: emptyList()
+
+            to.forEach {
+                toAutocompleteInput.setText(it)
+                contactAdapter.addFirstAvailableItem()
```

Comment:
Why call `addFirstAvailableItem` at every loop?

Reply:
To create the chips one by one for each recipient

Reply:
Might as well call "createChip" directly?

Reply:
Most of the content of the `addUnrecognizedMail` should indeed be called when adding these new recipients but I think that the email should be passed directly to a function as an argument rather than being obtained through the content of the `TextView` (first line of `addUnrecognizedMail`)

Reply:
So the fact of using directly the method sa I already thought of it during my analysis, but I found interesting to be able to recover the contact mail in the case that it existed, so use what existed. 
Although I didn't like this quick solution, but moreover when I look closely there is a bug on it, so it's like using the method directly 🥲. 
In this case go use the method directly.